### PR TITLE
Hook Plugin Registry up to Nomad, and plumb through a very basic integration smoke-test.

### PR DESCRIPTION
**Pull request by wimax-grapl**
### Which issue does this PR correspond to?
This is precursor work for https://app.zenhub.com/workspace/o/grapl-security/issue-tracker/issues/719
Just a very basic skeleton of a test framework, so that people can add real, valuable tests later.

### What changes does this PR make to Grapl? Why?
- Add a `nomad job validate` that would have unblocked an issue Colin had.
- Build plugin-registry with make build
- add plugin-registry as a service to nomad
- add a smoketest integration test for plugin-registry
- add a fake implementation of GetGeneratorsForEventSourceRequest so that we can run this integration test against something. 

### How were these changes tested?
make test-integration

### CODE REVIEW

File: src/rust/plugin-registry/src/client.rs [hunk 1]
```diff
@@ -33,6 +39,33 @@ pub struct PluginRegistryServiceClient<T> {
     inner: _PluginRegistryServiceClient<T>,
 }
 
+const HOST_ENV_VAR: &'static str = "GRAPL_PLUGIN_REGISTRY_HOST";
+const PORT_ENV_VAR: &'static str = "GRAPL_PLUGIN_REGISTRY_PORT";
+
+impl PluginRegistryServiceClient<Channel> {
```
Comment (wimax-grapl) [inline src/rust/plugin-registry/src/client.rs:45]: this is basically the same code as model-plugin-deployer/src/client.rs - our other extant rust grpc service

File: src/rust/plugin-registry/src/client.rs [hunk 2]
```diff
@@ -96,8 +129,8 @@ where
         self.inner
             .get_generators_for_event_source(GetGeneratorsForEventSourceRequestProto::from(request))
             .await
-            .expect("todo");
-        todo!()
+            .map(|resp| resp.into_inner().try_into().expect("proto to rs"))
+            .map_err(|_status| PluginRegistryServiceError::TodoImplementThisEnumError())
```
Comment (wimax-grapl) [inline src/rust/plugin-registry/src/client.rs:133]: i just needed some concrete PluginRegistryServiceError to get this to compile

File: src/rust/plugin-registry/src/server.rs [hunk 3]
```diff
@@ -135,18 +145,17 @@ impl PluginRegistryService for PluginRegistry {
...
 }
 
-pub async fn exec_service() -> Result<(), Box<dyn std::error::Error>> {
+pub async fn exec_service(socket_addr: SocketAddr) -> Result<(), Box<dyn std::error::Error>> {
     let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
     health_reporter
         .set_serving::<PluginRegistryServiceServer<PluginRegistry>>()
         .await;
 
-    let addr = "[::1]:50051".parse().unwrap();
```
Comment (wimax-grapl) [inline src/rust/plugin-registry/src/server.rs:144]: no more hardcoding! pass in the port

File: docker-compose.build.yml [hunk 4]
```diff
@@ -83,6 +83,15 @@ services:
       args:
         - RUST_BUILD=${RUST_BUILD:-debug}
 
+  plugin-registry:
```
[the remark this answers is not in the document]
Comment (wimax-grapl) [reply docker-compose.build.yml:86]: GOOD CATCH

[the remark this answers is not in the document]
Comment (wimax-grapl) [reply docker-compose.build.yml:86]: i might even automate this

File: src/rust/plugin-registry/src/server.rs [hunk 5]
```diff
@@ -124,7 +129,12 @@ impl PluginRegistryService for PluginRegistry {
         &self,
         _request: Request<GetGeneratorsForEventSourceRequestProto>,
     ) -> Result<Response<GetGeneratorsForEventSourceResponseProto>, Status> {
-        todo!()
+        // Stub implementation, for integration test smoke-test purposes.
+        // Replace with a real implementation soon!
+        let message = _request.into_inner();
+        Ok(Response::new(GetGeneratorsForEventSourceResponseProto {
+            plugin_ids: [message.event_source_id].into_iter().flatten().collect(),
+        }))
```
Comment (colin-grapl) [inline src/rust/plugin-registry/src/server.rs:137]: It might make sense to just leave this as `todo` and mark the test as `#[should_panic]` or something

**Review by colin-grapl — APPROVED** [review]

File: src/rust/plugin-registry/src/server.rs (hunk 5 above)
Comment (wimax-grapl) [reply src/rust/plugin-registry/src/server.rs:137]: _yes_ but I want to test that we get a real response. should_panic could result from any number of things, like the client not working. should_panic doesn't guarantee we have a real connection

Comment (wimax-grapl) [reply src/rust/plugin-registry/src/server.rs:137]: i think a response that's explicitly "yo this is a fake" would be the better solution

**Pull request merged**